### ISSUE TITLE
Switch Controls to new update system

### ISSUE THREAD

**Issue by RedstoneParadox** [RedstoneParadox/OakTree#13]
In order to clean up the `Control` update code, I made four new methods in `AbstractControl`:

1. `updateTree()`, which is run anytime the tree structure is updated to recreate the zIndexed list of controls that get updated every frame.
2. `interact()` which is where mouse-capture code is run, along with stuff such as click interactions.
3. `prepare()` is where anything else that needs to happen before drawing happens.
4. `draw()` is where drawing code is called.

Currently, all `Control` classes use `preDraw()` and `oldDraw()` to do all that, which is messy and calls code that does not need to run every frame.

Additionally, management of `Control` trees is being moved from `ControlGui` to `RootControl`, which manages the new system and makes things simpler by being a subclass of `AbstractControl`.

Finally, all the `InteractiveControl` stuff needs to be folded into `Control`.

- [x] `Control`
- [x] `SplitControl`
- [x] `ProgressBarControl`
- [x] `PaddingControl`
- [x] `PanelControl`
- [x] `ListPanelControl`
- [x] `PagePanelControl`
- [x] `GridPanelControl`
- [x] `BoxControl`
- [x] `InteractiveControl`*
- [x] `ButtonControl`
- [x] `HoverControl`
- [x] `DropdownControl`
- [x] `SliderControl`
- [x] `TextEditControl`
- [x] `SlotControl`
- [x] `EntityPreviewControl`
- [x] `LabelControl`

**Comment by RedstoneParadox** [comment]
It is done
